optimade/server/collections.py

```diff
@@ -10,21 +10,30 @@
 from optimade.filtertransformers.mongo import MongoTransformer
 
 from .models.util import NonnegativeInt
-from .models.modified_jsonapi import Resource
-from .models.structures import StructureMapper
+from .models.entries import EntryResource, EntryResourceAttributes
 from .deps import EntryListingQueryParams
 
+from .mappers.structures import StructureMapper
+
 
 config = ConfigParser()
 config.read(Path(__file__).resolve().parent.joinpath("config.ini"))
-RESPONSE_LIMIT = config["DEFAULT"].getint("RESPONSE_LIMIT")
+PAGE_LIMIT = config["DEFAULT"].getint("PAGE_LIMIT")
+PROVIDER = config["DEFAULT"].get("PROVIDER")
+PROVIDER_FIELDS = {field for field, _ in config["STRUCTURE"].items() if _ == ""}
```

Comment:
I don't fully agree with your inclusion of `provider*` etc. in the `MongoCollection`, since I feel it starts the Collection should not have to be initiated with knowledge of these things - they should rather be fed it from elsewhere. This is especially true for `page_limit`, which, e.g., has no meaning for single entry endpoints, and therefore does not need to be a Collection attribute.

However, I like that there is no `ConfigParser` in collections.py ... So I'm a bit in two minds over this, hehe.

Reply:
Hopefully this is resolved with my last commit, lmk if it's not what you had in mind

Reply:
It's brilliant, thanks!